src/maestro/mcptool.py

```diff
@@ -25,13 +34,27 @@ def create_mcptools(tool_defs):
 def create_mcptool(body):
     # Create an instance of the API class for the custom resource definition
     api_instance = client.CustomObjectsApi()
-    # Create the CRD instance
-    body["apiVersion"] = f"{group}/{version}"
-    body["kind"] = "MCPServer"
-    namespace = body["metadata"].get("namespace")
-    if not namespace:
-        namespace = "default"
-    api_response = api_instance.create_namespaced_custom_object(
-        group, version, namespace, plural, body
-    )
-    print(f"MCP tool: {api_response['metadata']['name']} successfully created")
+    url = body["spec"].get("url")
+    if url:
+        body["apiVersion"] = f"{remoteGroup}/{remoteVersion}"
+        body["kind"] = "RemoteMCPServer"
+        namespace = body["metadata"].get("namespace")
+        if not namespace:
+            namespace = "default"
+        api_response = api_instance.create_namespaced_custom_object(
+            remoteGroup, remoteVersion, namespace, remotePlural, body
+        )
+        print(
+            f"Remote MCP tool: {api_response['metadata']['name']} successfully created"
+        )
```

Comment:
There seems to be a lot of duplicate code here, I think there may be a simpler way to write this than an if else with almost the same code in each block.